docs/reference/aggregations/bucket/random-sampler-aggregation.asciidoc

```diff
@@ -0,0 +1,96 @@
+            "field": "taxful_total_price"
+          }
+        }
+      }
+    }
+  }
+}
+----
+// TEST[setup:kibana_sample_data_ecommerce]
+
+[[random-sampler-top-level-params]]
+==== Top-level parameters for random_sampler
+
+`probability`::
+(Required, float) The probability that a document will be included in the
+aggregated data. Must be greater than 0, less than `0.5`, or exactly `1`. The
+lower the probability, the fewer documents are matched.
+
+`seed`::
+(Optional, integer) The seed to generate the random sampling of documents. When
+a seed is provided, the random subset of documents is the same between calls.
+
+[[random-sampler-inner-workings]]
+==== How does the sampling work?
+
+The aggregation is a random sample of all the documents in the index. In other
+words, the sampling is over the background set of documents. If a query is
+provided, a document is returned if it is matched by the query and if the
+document is in the random sampling. The sampling is not done over the matched
+documents.
+
+Consider the set of documents `[1, 2, 3, 4, 5]`. Your query matches `[1, 3, 5]`
+and the randomly sampled set is `[2, 4, 5]`. In this case, the document returned
+would be `[5]`.
+
+This type of sampling provides almost linear improvement in query latency in relation to the amount
+by which sampling reduces the document set size.
+
+image::images/aggregations/random-sampler-agg-graph.png[Graph of the median speedup by sampling factor,align="center"]
+
+Above is typical of the speed up for the majority of aggregations for a test data set of 63 million documents. The exact constants will depend on the data set size and the number of shards, but the form of the relationship between speed up and probability holds widely. For certain aggregations, the speed up may not
+be as dramatic. These aggregations have some constant overhead unrelated to the number of documents seen. Even for
+those aggregations, the speed improvements can be significant.
+
+This sample set is generated by skipping documents using a geometric distribution
+(`(1-p)^(k-1)*p`) with success probability being the provided `probability` (`p` in distribution equation).
+The values returned from the distribution indicate how many documents to skip in
+the background. This is equivalent to selecting documents uniformly at random. It follows that the expected number of failures before a success is
+`(1-p)/p`. For example, with the `"probability": 0.01`, the expected number of failures (or
+average number of documents skipped) would be `99` with a variance of `9900`.
+Consequently, if you had only 80 documents in your index or matched by your
+filter, you would most likely receive no results.
+
+image::images/aggregations/relative-error-vs-doc-count.png[Graph of the relative error by sampling probability and doc count,align="center"]
+
+In the above image `p` is the probability provided to the aggregation, and `n` is the number of documents matched by whatever
```

Comment:
Not mandatory, but I think it would be nice to move this paragraph before the image, so that it provides context  in advance.

If you move this paragraph, it needs minor changes like this:
```suggestion
In the following graph, `p` is the probability provided to the aggregation, and `n` is the number of documents matched by whatever
```